### PR TITLE
Add validated input component

### PR DESCRIPTION
**Pull request by michaeljguarino**
Just simplifies reused logic around validating input fields we can use elsewhere

### CODE REVIEW

File: src/components/ValidatedInput.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+function ValidatedInputRef({label, hint, validation, onChange, width, ...input} : ValidatedInputProps, ref: Ref<any>) {
+    const [error, setError] = useState(null)
+    const wrappedOnChange = useCallback((e: any) => {
+        if (e.target?.value && validation) setError(validation(e.target.value))
+        if (onChange) onChange(e)
+    }, [onChange, validation])
+    
+    return (
+        <FormField label={label} hint={hint} caption={error?.message} width={width}>
```
Comment (dherault) [inline src/components/ValidatedInput.tsx:28]: I really think the error message should be with the hint not the caption.

Comment (dherault) [reply src/components/ValidatedInput.tsx:28]: Also the files are not linted sowwy :)

File: src/stories/ValidatedInput.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    />
+  )
+}
+
+export const Default = Template.bind({})
+
+Default.args = {
+  user: {
+    name: 'Jane Smith',
+    email: 'jane.smith@plural.sh',
```
Comment (dherault) [inline src/stories/ValidatedInput.stories.tsx:28]: This should be removed

**Review by dherault — CHANGES_REQUESTED** [review]

File: src/components/ValidatedInput.tsx (hunk 1 above)
Comment (michaeljguarino) [reply src/components/ValidatedInput.tsx:28]: That's probably true, will the hint text be highlighted with an error color though?

Comment (dherault) [reply src/components/ValidatedInput.tsx:28]: Hint can accept a ReactNode, so it's up to you to color it

Comment (michaeljguarino) [reply src/components/ValidatedInput.tsx:28]: Although would this look weird with interfaces with multiple inputs all being validated? Caption might be better there

**Comment by michaeljguarino** [comment]
going to merge and keep chugging, we can change if need be

**Pull request merged**